### PR TITLE
UI: add live-updating to exec sidebar

### LINKED ISSUE #7454: UI: add live-updating for exec sidebar

### PR DESCRIPTION
**Pull request by backspace**
This closes #7454. It makes use of the existing watchable tools to
allow the exec popup sidebar to be live-updating. It also adds
alphabetic sorting of task groups and tasks.

sorted-live-updating-sidebar

### CODE REVIEW

**Comment by backspace** [comment]
I noticed a curious bug while making a GIF for the blog post, where the active task indicator switches tasks. I’d move this to Draft if I could! I’ll hopefully be able to fix the bug next week.

sidebar-blog

File: ui/app/controllers/exec.js [hunk 1]
```diff
@@ -20,9 +21,17 @@ export default Controller.extend({
   socketOpen: false,
   taskState: null,
 
-  runningAllocations: filterBy('model.allocations', 'isRunning'),
-  runningTaskGroups: mapBy('runningAllocations', 'taskGroup'),
-  uniqueRunningTaskGroups: uniq('runningTaskGroups'),
+  pendingAndRunningAllocations: computed('model.allocations.@each.clientStatus', function() {
+    return this.model.allocations.filter(
+      allocation => allocation.clientStatus == 'pending' || allocation.clientStatus == 'running'
```
Comment (DingoEatingFuzz) [inline ui/app/controllers/exec.js]: loose equality? 😱

Comment (backspace) [reply ui/app/controllers/exec.js]: hmm indeed, I’m surprised there wasn’t a lint error on that. Fixed, thanks. I’m still trying to fix the bug I mentioned in my latest comment, though, sadly.

**Comment by backspace** [comment]
This is ready for rereview. To address the bug shown in that GIF I ended up changing the controller’s `taskState` property from something set when the task is chosen in the sidebar to a computed property that’s responsive to allocations being destroyed as updates come in. Having the `taskState` be calculated once was unsuitable because when a job has been updated, a whole new set of states will be created, so making it a computed property that derives the state from the task group and task names makes it resilient across job versions.

There’s still a bit more work to come on this trajectory because there’s currently no handling for a job becoming dead or even already being dead when the window opens (#7456). Also, if a task state stops being `isActive`, it won’t show in the sidebar, but nothing will change in the main terminal area if that task state has already been selected; it will still look like you can open an exec session. PR for these cases forthcoming.

File: ui/app/controllers/exec.js [hunk 2]
```diff
@@ -35,14 +43,53 @@ export default Controller.extend({
...
+
+  taskState: computed(
+    'allocations.[]',
+    'allocationShortId',
+    'allocations.@each.isActive',
+    'taskName',
+    'taskGroupName',
+    'allocation',
+    'allocation.states.@each.name',
+    'allocation.states.@each.isRunning',
```
Comment (DingoEatingFuzz) [inline ui/app/controllers/exec.js:56]: Lol, so many dependent keys. Gotta love our data model.

Comment (backspace) [reply ui/app/controllers/exec.js:56]: I had an intermediate `allocation` computed property that would have reduced this somewhat but it didn’t work for whatever reason and I gave up on it  😞

**Pull request merged**